src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -232,16 +238,35 @@ private String queryContainerSampleFileRawData(List<QCMetricConfiguration> confi
     }
 
     /** Prefer the standalone variant of a metric if it's also part of a paired config so that we avoid double-counting */
-    private void retainIfPreferred(Map<String, QCMetricConfiguration> preferredConfigs, QCMetricConfiguration configuration, String label)
+    private void retainIfPreferred(Map<String, QCMetricConfiguration> preferredConfigs, QCMetricConfiguration configuration, String label, List<QCMetricConfiguration> configurations, boolean forOutliers)
     {
         QCMetricConfiguration existingConfig1 = preferredConfigs.get(label);
         if (existingConfig1 == null || existingConfig1.getSeries2Label() == null)
         {
-            preferredConfigs.put(label, configuration);
+            if (forOutliers)
+            {
+                if (configuration.getName().equalsIgnoreCase(label))
+                {
+                    preferredConfigs.put(label, configuration);
+                }
+                else
+                {
+                    configurations.forEach(c -> {
```

Comment:
This loop makes it $O(N^2)$. I am open to suggestions to avoid this.

Reply:
We could switch to passing in a Map (label->config object) with all of the configurations instead of a List.

Reply:
I can check on this but I think that will destroy the original intent of creating this map to avoid the double counting of the outliers for those paired metric configs.

Reply:
This traversal will only happen for the paired metric configs which are just a couple at the moment.

Reply:
Sorry, I meant switch `configurations` from a List to a Map, not `preferredConfigs` from a Map to a List.

But regardless, given the small number of metrics and the even smaller number that will be subject to this code, it's unlikely to be noticeable in practice.